odl/tomo/backends/util.py

```diff
@@ -0,0 +1,26 @@
+__all__ = tuple()
+
+
+def _add_default_complex_impl(fn):
+    """Wrapper to call a class method twice when it is complex."""
+
+    def wrapper(self, x, out, **kwargs):
+        if self.vol_space.is_real and self.proj_space.is_real:
+            fn(self, x, out, **kwargs)
```

Comment:
You're correct, `self` has to be supplied. Same for the complex case below.

Reply:
Sharp! Real-time feedback :)

Reply:
Yes, but now it's late enough and my sharpness declines rapidly :-)